__tests__/server.test.js

```diff
@@ -39,3 +39,41 @@ describe('GET /api/topics',()=>{
     })
 })
 
+describe('GET /api/articles/:article_id',()=>{
+    test('should return a 200 status code when given a valid id',()=>{
+        return request(app)
+        .get('/api/articles/4')
+        .expect(200)
+    })
+    test('should return the requested article',()=>{
+        return request(app)
+        .get('/api/articles/4')
+        .expect(200)
+        .then(({body})=>{
+            expect(body.article).toMatchObject({
+                    title: "Student SUES Mitch!",
+                    topic: "mitch",
+                    author: "rogersop",
+                    body: "We all love Mitch and his wonderful, unique typing style. However, the volume of his typing has ALLEGEDLY burst another students eardrums, and they are now suing for damages",
+                    created_at: '2020-05-06T01:14:00.000Z',
+                    article_img_url:"https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+            })
+        })
+    })
+    test('should return a 404 with the message article doesn\'t exist if given an id that does not exist in the database',()=>{
+        return request(app)
+        .get('/api/articles/999')
+        .expect(404)
+        .then(({body})=>{
+            expect(body.message).toBe("Article doesn't exist")
+        })
+    })
+    test('should return a 404 message Invalid id if given an invalid id',()=>{
+        return request(app)
+        .get('/api/articles/notAnId')
+        .expect(404)
+        .then(({body})=> {
+            expect(body.message).toBe("Invalid id")
+        })
+    })
+})
```

Comment:
This test for 'invalid id' should be a 400, 404 would be valid but non-existant.